pages/[id].jsx

```diff
@@ -1,19 +1,40 @@
+import { useEffect, useState } from 'react';
 import { useRouter } from 'next/router';
 import Head from 'next/head';
 
+import data from '../lib/data';
+
 export default function DetailPage() {
+  const [teamMemberDetails, setTeamMemberDetails] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
A boolean works for simple cases, but often there are more states than we realize. For example, this should also have an error state in case the data fails to load.

What I like to do is set up an object like:
```js
const loadingStates = {
  EMPTY: 'empty',
  LOADING: 'loading',
  LOADED: 'loaded',
  ERROR: 'error',
};
```

Then we can use state to update it with more flexibility for adding different UI states:
```js
const [loadingState, setLoadingState] = useState(loadingStates.EMPTY);

// later, if there's an error
setLoadingState(loadingStates.ERROR);

// rendering the UI state
{loadingState === loadingState.ERROR && <ErrorState />}
```